kernel/src/path.rs

```diff
@@ -566,4 +603,34 @@ mod tests {
         assert!(matches!(log_path.file_type, LogPathFileType::Commit));
         assert_eq!(log_path.filename, "00000000000000000010.json");
     }
+
+    #[test]
+    fn test_new_uuid_parquet_checkpoint() {
```

Comment:
maybe worth having a check that makes sure that everything matches in to_string etc.? and maybe just scrub the uuid part since it's known length?